[DX-955] Add use-case based SDK references across Pub/Sub pages (#3340)
ably/docs: src/pages/docs/spaces/index.mdx

* [Authentication](/docs/auth) and [connection management](/docs/connect) are handled by the underlying SDK.
* You can use the Ably JavaScript SDK to extend your application's functionality further with [Pub/Sub](/docs/basics).

### Enhance collaborative applications with other Ably products

Ably provides purpose-built SDKs for common realtime features that complement the Spaces SDK:

> * **[Ably Chat](/docs/chat)**: Add chat, typing indicators, reactions, and message history to your collaborative application using the Chat SDK rather than building these features from scratch.

Reviewer: `rather than building these features from scratch` reads as preachy — the link to the Chat SDK already implies the recommendation. Project writing-style-guide also calls out avoiding bold prefixes in bullets, but other Spaces pages use them, so I've kept the prefix here:

```suggestion
* **[Ably Chat](/docs/chat)**: Add chat, typing indicators, reactions, and message history with the Chat SDK.
```

Author: Keeping "rather than building these features from scratch". The link alone implies the recommendation, but the explicit phrasing is the value prop a Spaces reader is weighing here — they're deciding whether to adopt the Chat SDK or hand-roll chat on Pub/Sub. Spelling that out is worth the few extra words.

Reviewer 2: Wary of adding this when we haven't tested these working together.

Author: Sticking to what @kennethkalmer said 👍

Author: https://github.com/ably/docs/pull/3340/changes/fe2d1956260957f50ade0818f3ea54c62f772eb0

> * **[Pub/Sub](/docs/basics)**: Use Pub/Sub for custom features not covered by a purpose-built SDK, such as notifications, 'follow me' and 'bring to me' functionality, or custom application-specific events.

Reviewer: `custom` repeats (`custom features... custom application-specific events`). Drop the second:

```suggestion
* **[Pub/Sub](/docs/basics)**: Use Pub/Sub for features not covered by a purpose-built SDK, such as notifications, 'follow me' and 'bring to me' functionality, or application-specific events.
```

Author: Keeping the second `custom`. The two uses aren't redundant: the first `custom features` describes the general category (anything outside the purpose-built SDKs), the second `custom application-specific events` distinguishes user-defined event types from the named patterns (`notifications`, `follow me`, `bring to me`) listed before it.
